Lib/bdb.py

```diff
@@ -155,6 +157,9 @@ def dispatch_return(self, frame, arg):
         if self.stop_here(frame) or frame == self.returnframe:
             # Ignore return events in generator except when stepping.
             if self.stopframe and frame.f_code.co_flags & GENERATOR_AND_COROUTINE_FLAGS:
+                # It's possible to trigger a StopIteration exception in
+                # the caller so we must set the trace function in the caller
+                self._set_caller_tracefunc(frame)
```

Comment:
how is this related to this PR?

Reply:
There is a hidden bug which was exposed by the change of `break_anywhere`. There's a test case of `bdb` testing raising `StopIteration` in a generator. It passed because `break_anywhere` always return `True` when the function was defined in the same file, which always sets the trace function on the caller because it is in the same file. That coincidence hide the bug where returning from a generator should stop in the caller. There's a similar call in line 177 - that's for the normal return case from https://github.com/python/cpython/pull/118979 - it's also my fix to a similar case.